commit d670fb3473fbb455881125086f649e416ea755fe
Author: Chris Liddell <chris.liddell@artifex.com>
Date:   Wed Oct 12 13:36:22 2011 +0100

    Bug 689450: deal with invalid font entry in res dict.
    
    In the event that an entry in a resource dictionary for a font is or
    references an invalid object type (i.e. not a dictionary), issue a
    warning and use the same logic as we do for a missing font resource.
    
    For this error, we'll also honor the PDFSTOPONERROR setting.
    
    In addition, as I was touching the logic around the same place, this
    commit changes the missing font resource case to also honor PDFSTOPONERROR.
    
    No cluster differences expected.

diff --git a/gs/Resource/Init/pdf_font.ps b/gs/Resource/Init/pdf_font.ps
index f70798e..04ba043 100644
--- a/gs/Resource/Init/pdf_font.ps
+++ b/gs/Resource/Init/pdf_font.ps
@@ -1937,13 +1937,39 @@ drawopdict begin
   } bdef
   /Tf {
     1 index Page /Font rget {
-      resourcefont exch Tf pop
+      dup type /dicttype eq {
+        resourcefont exch
+        Tf pop
+      }
+      {
+        PDFSTOPONERROR
+        {
+          /typecheck cvx signalerror
+        }
+        {
+          % Bug 689450
+          type
+          (   **** Warning: Tf refers to a resource key with an invalid value type: ) pdfformaterror
+          .namestring pdfformaterror
+          (. Assuming resource key: ) pdfformaterror
+          1 index .namestring pdfformaterror
+          ( is a font name.\n) pdfformaterror
+          Tf
+        } ifelse
+      }ifelse
     }
-    { % Bug 689037
-      (   **** Warning: Tf refers to an unknown resource name: ) pdfformaterror
-      1 index .namestring pdfformaterror
-      ( Assuming it's a font name.\n) pdfformaterror
-      Tf
+    {
+      PDFSTOPONERROR
+      {
+          pop /undefined cvx signalerror
+      }
+      {
+        % Bug 689037
+        (   **** Warning: Tf refers to an unknown resource name: ) pdfformaterror
+        1 index .namestring pdfformaterror
+        ( Assuming it's a font name.\n) pdfformaterror
+        Tf
+      } ifelse
     } ifelse
   } bdef
 end
